tasks/destroy-volumes.yml

```diff
@@ -4,8 +4,9 @@
     destroy_virt_volume.sh
     {{ item.name }}
     {{ item.pool }}
-  with_items: "{{ libvirt_vm_volumes }}"
+  with_items: "{{ vm.volumes | default([]) }}"
```

Comment:
I'm not sure that this default will work in the same way as before. The default filter only applies the default if the variable is undefined. I think there are two potential solutions. Either omit the deprecated variables from defaults/main.yml, such that they are actually undefined, or use the boolean argument to default. Possibly the first is more consistent.

See http://jinja.pocoo.org/docs/2.10/templates/#default.

Reply:
Same goes for other defaults.

Reply:
Good point. I don't think omitting the deprecated variables from `defaults/main.yml` will work, because they are being referenced in the default singleton definition of `libvirt_vms`, and we get an error of referencing an undefined variable. Even if we use default(omit) filters in each element of the `libvirt_vms` definition, [omitting role variables doesn't work](https://github.com/ansible/ansible/issues/22417).

I guess using default's boolean argument is the best option.

Reply:
Having said all that, using your vars suggestion means we don't need the default filters at all!

Reply:
Did a little testing and looks like you're right. Don't we need to use default when defining the vars though?